### ISSUE TITLE
get_glyphs() rusty interface

### ISSUE THREAD

**Issue by mersinvald** [rise-ui/raqm-rs#2]
There is a `get_glyphs()` raqm method which is returning a borrowed pointer to an array of `raqm_glyph_t` elements.

Though it is easy to transform it into a rust slice, I can't find it ergonomic to expose bare c types in the public interface.

So we either need to transform this array into a Vec of rust types, which will cause us a memory allocation, or return an iterator from the `get_glyphs()` function.

The difference in usage is self-explanatory, so I need your input on that, @friktor, as I have no idea how this method is usually used and if the iterator approach (which is both easier to implement and zero-cost) is feasible for the common usage of the lib.

Summing up, we will have rust wrapper type anyway:
```rust
struct Glyph<'a> {
    inner: &'a raqm_glyph_t
}

impl<'a> Glyph<'a> {
    /* getters and setters */
}
```

And we need to decide between
```rust
fn get_glyphs<'a>(&'a mut self) -> Result<Vec<Glyph<'a>>>;
```
And 
```rust
fn get_glyphs<'a>(&'a mut self) -> Result<impl Iterator<Item = Glyph<'a>>>
```

**Comment by friktor** [comment]
@mersinvald It's better to use ```fn get_glyphs<'a>(&'a mut self) -> Result<impl Iterator<Item = Glyph<'a>>>``` because on top of this module i will write own converter to webrender with its own builder module .

**Comment by mersinvald** [comment]
@friktor then iterator it is.

**Comment by mersinvald** [comment]
@friktor meh, after looking into the libraqm code, I found out that we have no way to ensure the validity of references and that the memory has not been freed yet through the type system.

So I'm falling back to Vec<Glyph> variant, you'll get to manually transform the data to whatever you'll need within webrender.
If this memory alloc will come to be a bottleneck, we'll deal with it later.
